src/Kernel/ProtoObject.class.st

```diff
@@ -67,9 +67,11 @@ ProtoObject >> becomeForward: otherObject copyHash: copyHash [
 ProtoObject >> cannotInterpret: aMessage [
 	 "Handle the fact that there was an attempt to send the given message to the receiver but a null methodDictionary was encountered while looking up the message selector.  Hopefully this is the result of encountering a stub for a swapped out class which induces this exception on purpose."
 
-"If this is the result of encountering a swap-out stub, then simulating the lookup in Smalltalk should suffice to install the class properly, and the message may be resent."
+	"If this is the result of encountering a swap-out stub, then simulating the lookup in Smalltalk should suffice to install the class properly, and the message may be resent."
+	
+	"Important: do not send anything to self (here or in a redefinition) as that will likely cause an infinite recursion of `cannotInterpret:`"
 
-	(self class lookupSelector: aMessage selector) ifNotNil:
+	((Behavior classOf: self) lookupSelector: aMessage selector) ifNotNil:
```

Comment:
But this restarts the lookup from the receiver's class. I don't think that  `lookupSelector:` works if the method dictionary is nil. In that case you may have an exception, and the exception may want to print the objects in the stack, and then it will try to print your proxy. And recurse.

Reply:
I checked it. ` lookupSelector:` is naive and just call `methodDict at:` without any check.
But `methodDict` is a robust accessor that returns an empty method dictionary if the ivar is nil. (it's documented to be lenient with tools that do not expect nil as an answer)
So the `lookupSelector:` should work and just look up until ProtoObject, find nothing, and return nil.

Then the next step in `ProtoObject>>cannotInterpret:` is just to `Error signal: 'MethodDictionary fault'.`.
That should be caught by `should:raise:` in the test method, but is not according to Jenkins: https://ci.inria.fr/pharo-ci-jenkins2/blue/organizations/jenkins/Test%20pending%20pull%20request%20and%20branch%20Pipeline/detail/PR-14211/3/tests

My hypothesis is that the stack, context, pc or something internal is inconsistent.